src/xrpld/app/tx/applySteps.h

```diff
@@ -29,6 +29,13 @@ class Application;
 class STTx;
 class TxQ;
 
+struct ApplyResult
+{
+    TER ter;
+    bool applied;
+    std::optional<TxMeta> metadata = std::nullopt;
```

Comment:
This one is unnecessary  as default constructor will do the same.

Reply:
I added a custom constructor instead to ensure the other values are defined. a897804f389d34495eaec1457455c2e236b7ebcb